### ISSUE TITLE
Error: UNHANDLED REJECTION write EPIPE

### ISSUE THREAD

**Issue by jack0wsky** [gatsbyjs/gatsby#28011]
## Description

Hello everyone!

I'm getting this error almost every time I shut down local server.
```
error UNHANDLED REJECTION write EPIPE


  Error: write EPIPE
  
  - child_process.js:806 ChildProcess.target._send
    internal/child_process.js:806:20
  
  - child_process.js:676 ChildProcess.send
    internal/child_process.js:676:19
  
  - develop.ts:130 ControllableScript.stop
    [ap-investment]/[gatsby]/src/commands/develop.ts:130:20
  
  - develop.ts:500 shutdownServices
    [ap-investment]/[gatsby]/src/commands/develop.ts:500:28
  
  - develop.ts:442 process.<anonymous>
    [ap-investment]/[gatsby]/src/commands/develop.ts:442:11

```
I've tried multiple times
```
gatsby clean
remove /node_modules and package-lock.json
npm install
```
but it solves problem only for some time. Don't know if it is related but when ```gatsby develop``` and dev server is running, after some saves, when I click Command + S to save changes, hot reload doesn't work and project isn't re-rendering.
### Expected result

Shuting down dev server without errors

### Actual result

Throwing error mentioned above and hot reload not working.

### Environment

```
  System:
    OS: macOS 10.15.7
    CPU: (4) x64 Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.14.0 - /usr/local/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.8 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 86.0.4240.193
    Firefox: 82.0.3
    Safari: 14.0
  npmPackages:
    gatsby: ^2.25.3 => 2.25.4 
    gatsby-alias-imports: ^1.0.6 => 1.0.6 
    gatsby-image: ^2.4.21 => 2.4.21 
    gatsby-plugin-postcss: ^3.0.4 => 3.0.4 
    gatsby-plugin-sass: ^2.4.2 => 2.4.2 
    gatsby-plugin-sharp: ^2.7.1 => 2.7.1 
    gatsby-plugin-tinacms: ^0.32.0 => 0.32.0 
    gatsby-source-filesystem: ^2.4.2 => 2.4.2 
    gatsby-tinacms-git: ^0.32.0 => 0.32.0 
    gatsby-tinacms-json: ^0.32.0 => 0.32.0 
    gatsby-tinacms-remark: ^0.32.1 => 0.32.1 
    gatsby-transformer-remark: ^2.9.2 => 2.9.2 
    gatsby-transformer-sharp: ^2.5.21 => 2.5.21 
  npmGlobalPackages:
    gatsby-cli: 2.12.117
    gatsby: 2.25.3

```

**Comment by jamo** [comment]
@jack0wsky try updating to a more recent version of gatsby (as of writing this we are at 2.28.0). 
Not fully sure if the underlying issue was fully fixed but I haven't run into this for a while with a fresh gatsby version.

**Comment by hronchetti** [comment]
> @jack0wsky try updating to a more recent version of gatsby (as of writing this we are at 2.28.0).
> Not fully sure if the underlying issue was fully fixed but I haven't run into this for a while with a fresh gatsby version.

Worked for me!

**Comment by aravind-tc** [comment]
We are still seeing this issue with the latest gatsby

 ERROR 

UNHANDLED REJECTION write EPIPE



  Error: write EPIPE
  
  - child_process.js:806 ChildProcess.target._send
    internal/child_process.js:806:20
  
  - child_process.js:676 ChildProcess.send
    internal/child_process.js:676:19
  
  - develop.ts:130 ControllableScript.stop
    [bigfig]/[gatsby]/src/commands/develop.ts:130:20
  
  - develop.ts:500 shutdownServices
    [bigfig]/[gatsby]/src/commands/develop.ts:500:28
  
  - develop.ts:442 process.<anonymous>
    [bigfig]/[gatsby]/src/commands/develop.ts:442:11

  npmPackages:
    gatsby: ^2.26.1 => 2.26.1
    gatsby-image: ^2.5.0 => 2.5.0
    gatsby-plugin-manifest: ^2.6.1 => 2.6.1
    gatsby-plugin-offline: ^3.4.0 => 3.4.0
    gatsby-plugin-react-helmet: ^3.4.0 => 3.4.0
    gatsby-plugin-sharp: ^2.8.0 => 2.8.0
    gatsby-plugin-styled-components: ^3.8.0 => 3.8.0
    gatsby-source-filesystem: ^2.5.0 => 2.5.0
    gatsby-transformer-sharp: ^2.6.0 => 2.6.0
  npmGlobalPackages:
    gatsby-cli: 2.7.29

**Comment by karlhorky** [comment]
Yes, this error also happens for me every time I shut down using <kbd>ctrl</kbd>-<kbd>C</kbd> on macOS.

Demo repo:

https://github.com/upleveled/gatsby-serverside-auth0/tree/main/packages/gatsby-website

**Comment by sol-idsnake** [comment]
Same thing happens on Win10:

```
    "gatsby": "^2.31.1",
    "gatsby-image": "^2.10.0",
    "gatsby-plugin-google-analytics": "^2.10.0",
    "gatsby-plugin-react-helmet": "^3.9.0",
    "gatsby-plugin-sass": "^3.1.0",
    "gatsby-plugin-sharp": "^2.13.4",
    "gatsby-plugin-sitemap": "^2.11.0",
    "gatsby-source-filesystem": "^2.10.0",
    "gatsby-source-wordpress-experimental": "^7.0.3",
    "gatsby-transformer-sharp": "^2.11.0",
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "react-helmet": "^6.1.0",
```

`"develop": "gatsby develop -H=0.0.0.0",`

Node: v14.15.4
npm: 6.14.10

**Comment by siemah** [comment]
hey @jack0wsky , @karlhorky , @sol-idsnake  and @aravind-tc there is nothing wrong with GatsbyJS its how NodeJS handle piping data into closed streams, the error might be caused when the NodeJS event loop didn't clean-up closed TCP connections from the HTTP connection pool and then the NodeJS runtime attempted to use the closed TCP connection.

**Comment by karlhorky** [comment]
Hmm... but I think there's a way to hook into the `SIGINT` event that is triggered by `ctrl-C` and prevent these types of things. I remember fixing a similar issue with logging output from a process not stopping on `SIGINT` a few years ago. I think this can be fixed in Gatsby itself.

**Comment by karlhorky** [comment]
hello stalebot

**Comment by thelovesmith** [comment]
Has anyone figured out any solutions for this error? IM getting the same EPIPE error every time I shut down my local server.

**Comment by Jim-Horn** [comment]
I almost always get that same error. I'm using `iTerm` & `zsh`. Wondering if it's related to either of those.

**Comment by rptomey** [comment]
I get this error, too. Happens every time I use control+c to shut down gatsby develop mode.

**Comment by karlhorky** [comment]
not stale

**Comment by karlhorky** [comment]
not stale

**Comment by jamo** [comment]
Which version of gatsby are you using @karlhorky ?

**Comment by karlhorky** [comment]
Always latest: `3.8.0` currently

**Comment by karlhorky** [comment]
confirmed still happens on `3.8.0`:

<img width="473" alt="Screen Shot 2021-06-29 at 14 58 00" src="https://user-images.githubusercontent.com/1935696/123801459-99d8c800-d8ea-11eb-8663-1a9d38171ce6.png">

**Comment by marcinczenko** [comment]
and also 3.9.0...

**Comment by schabibi1** [comment]
I have the same issue in both cases before update to `3.9.0` and after.
This happens with starters or without.

But somehow, the projects which I created about a month ago with `^3.6.2` work with no error logs after `ctrl + c` .
The only difference I made is that the Gatsby version.

**Comment by karlhorky** [comment]
@schabibi1 I'm not sure if this is what you are experiencing, but I've found this to be an intermittent problem (it doesn't always happen). Sometimes I <kbd>ctrl</kbd>-<kbd>c</kbd> and there is no error, but often there is the error. Maybe if you try starting and stopping a bunch of times, you will see that the `^3.6.2` version is also affected...

**Comment by schabibi1** [comment]
@karlhorky I tried to run and terminate several times but so far, I just get an EPIPE error.
I mean, there are no errors while running so, technically, I just can ignore them.
I just don't want to get used to ignoring the errors and warnings😅

**Comment by karlhorky** [comment]
> I just don't want to get used to ignoring the errors and warnings😅

Yeah, definitely! Feel the same myself. Every time that Gatsby shows this error, I think "oh no, have to debug something here" for a half second before remembering

**Comment by KyleAMathews** [comment]
Looks like we just need to add a try/catch around this [line](https://github.com/gatsbyjs/gatsby/blob/54ed0c301631450720e341a33ebc7ffce99e5528/packages/gatsby/src/commands/develop.ts#L130). Apparently a child process of Gatsby sometimes crashes before we try to explicitly kill it which causes an error as we can no longer write to it.

Would someone line to PR a fix? This is an irritating warning indeed!

**Comment by karlhorky** [comment]
@KyleAMathews as simple as this? https://github.com/gatsbyjs/gatsby/pull/32311

**Comment by ChrisSargent** [comment]
FYI - might help some people troubleshooting. I have a monorepo with multiple Gatsby apps.

One of our apps was using `"gatsby": "^3.8.1"` and the others were on `"gatsby": "^3.7.2"`. Setting them all to `^3.7.2` prevented this issue.

Hence it was either being caused by having mis-matched versions or something that was introduced between `3.7.2` and `3.8.1`

**Comment by karlhorky** [comment]
Hmm... but what was the exact version (in the `package-lock.json` or `yarn.lock` files)? `^3.7.2` will actually get you the latest version under `4.0.0`...

I was experiencing this long before version 3, and haven't seen a non-broken version yet...

**Comment by ChrisSargent** [comment]
My gut feeling is that it was related to mixed dependencies in my case. Will try to check the locked versions.

**Comment by karlhorky** [comment]
Thanks for the merge @vladar ! 

Which version of which package should I upgrade to in order to test this?

**Comment by karlhorky** [comment]
Ah I guess there's also #32356 to wait on. Will keep an eye on that too.

**Comment by vladar** [comment]
@karlhorky I think the fix will go out on Tuesday in the next minor (3.10.0) as a part of the release process. But in the meantime, you can try it using `yarn add gatsby@next` which points to the latest unpublished version.

**Comment by ChrisSargent** [comment]
FYI, this was what was in my yarn.lock file `gatsby@^3.7.2, gatsby@^3.9.0: version "3.9.1"`

**Comment by herecydev** [comment]
Still experiencing this on gatsby 3.10.1

![Untitled](https://user-images.githubusercontent.com/11328618/126814237-9ebc8797-b320-41b1-8926-cf4c87d5815b.png)

**Comment by karlhorky** [comment]
@KyleAMathews @vladar can we reopen this one?

**Comment by iskandar47** [comment]
I'm getting this error after upgrading from v2 to v3

**Comment by jjercx** [comment]
> I'm getting this error after upgrading from v2 to v3

Same here, and I also lost hot reloading

**Comment by Danishkhurshid** [comment]
Getting this error after upgrading from v2 to v3.

**Comment by tomwrankmore** [comment]
This error came out of nowhere half way throught building a project, it's anxiety inducing to see red in the Terminal all day.

**Comment by jamo** [comment]
@tomwrankmore are you on the latest gatsby version? I believe the fix was fixed recently

**Comment by herecydev** [comment]
> Still experiencing this on gatsby 3.10.1

@jamo are you on this version?

**Comment by tomwrankmore** [comment]
> @tomwrankmore are you on the latest gatsby version? I believe the fix was fixed recently

I'm on the latest Gatsby version: 3.10.2 yeah! Weird...

**Comment by yassine-mokni** [comment]
the same issue even in the latest version @3.10.2

**Comment by Geerodge** [comment]
Getting this error after upgrading from v2 to v3, using v3.10.2.

**Comment by zqcccc** [comment]
> Getting this error after upgrading from v2 to v3, using v3.10.2.
> ![Untitled](https://user-images.githubusercontent.com/4022132/127672254-5820a6e2-e857-43cf-94f8-2695a3ddda6e.png)

same with you

**Comment by Geerodge** [comment]
Is there a solution to this that I've missed, as this is closed?

**Comment by jacurtis** [comment]
This was closed with a previous version of Gatsby and it is now back with the most recent version. So it needs to be reopened.

**Comment by sniperyork** [comment]
Can confirm that I'm seeing the same issue in v3.10.2

**Comment by liminalfun** [comment]
Hi, I gotten the same issue every time I've tried starting a new project from [Gatsby's default starter](https://github.com/gatsbyjs/gatsby-starter-default). I tried three or four times between from June until now. 

I've had a consistent bonus issue too tho ;) Something about "cannot find autoprefixer". 

I've tried multiple times: 
gatsby clean
remove /node_modules and package-lock.json
npm install

Rinse, repeat, give up on Gatsby 🤷‍♀️

I'm basically done trying Gatsby at this point, but am posting here just in case someone else runs into the same issue ! Take a break and try something else :)

**Comment by LazKot** [comment]
i can confirm i'm seeing this same issue in v3.10.2
![Screenshot 2021-08-05 124839](https://user-images.githubusercontent.com/38168706/128283113-21ad2f19-8a99-4c54-9772-18fb41703fb3.png)

**Comment by gh640** [comment]
This occurs with v3.11.0.

> i can confirm i'm seeing this same issue in v3.10.2
> ![Screenshot 2021-08-05 124839](https://user-images.githubusercontent.com/38168706/128283113-21ad2f19-8a99-4c54-9772-18fb41703fb3.png)

`develop.ts:540`:

```typescript
    telemetryServerProcess.stop(),
```

https://github.com/gatsbyjs/gatsby/blob/gatsby%403.11.0/packages/gatsby/src/commands/develop.ts#L540

`develop.ts:133` is the line `this.process.send({` in `stop()`:

```typescript
  async stop(
    signal: NodeJS.Signals | null = null,
    code?: number
  ): Promise<void> {
    if (!this.process) {
      throw new Error(`Trying to stop the process before starting it`)
    }

    this.isRunning = false
    try {
      if (signal) {
        this.process.kill(signal)
      } else {
        this.process.send({
          type: `COMMAND`,
          action: {
            type: `EXIT`,
            payload: code,
          },
        })
      }
    } catch (err) {
      // Ignore error if process has crashed or already quit.
      // Ref: https://github.com/gatsbyjs/gatsby/issues/28011#issuecomment-877302917
    }
```

https://github.com/gatsbyjs/gatsby/blob/gatsby%403.11.0/packages/gatsby/src/commands/develop.ts#L120-L145

It seems `telemetryServerProcess` doesn't stop properly with `send()` is invoked.

I don't know well about the `execa` process used here but tried changing `this.process.send({ ... })` to `this.process.cancel()` and the error has gone. I hope this helps.

**Comment by LekoArts** [comment]
Third time's the charm: https://github.com/gatsbyjs/gatsby/pull/32744
Once merged will be backported and released in `gatsby@3.11.1`

**Comment by gh640** [comment]
I confirmed the error has gone with `gatsby@3.11.1`. Thanks!

**Comment by Geerodge** [comment]
I can confirm the error has gone with gatsby@3.11.1 as well. Nice one 👍

**Comment by LazKot** [comment]
can also confirm error has been fixed in 3.11.1. Thanks all!

**Comment by GianaS** [comment]
I'm getting this same issue with Gatsby v.4.0.2

**Comment by Piligrimas** [comment]
Getting this error on Gatsby v4.3.0

**Comment by colinrim** [comment]
Netlify builds with Gatsby v.4.3.0 or gatsby@next all failing with this error.

**Comment by unimprobable** [comment]
I am having this error after upgrading Gatsby from 3.14 to 4.4, deploying on Netlify.

Netlify seems to think it is a memory error, that the build exceeds the 3GB allotted for builds.

**Comment by jack0wsky** [comment]
Same error during building and deploying on Netlify in Gatsby 4.5.0-next.3 after upgrade from 3.11.1

**Comment by NeversSync** [comment]
Having the same error on about 1/5th of build in Netlify on Gatsby 4.0.1

**Comment by panzacoder** [comment]
Same error intermittently on Netlify with no explanation. It seems from this I need to revert back to Gatsby v3, which is a big bummer.

**Comment by shalawfatah** [comment]
same error with Gatsby 4.3.0, I'm sure I will have the same error even if I update to 4.6.0.
``` gatsby develop ``` works well, but ``` gatsby build ``` fails every time with this error.

**Comment by kb1995** [comment]
Not sure why this issue has been closed since I still get it for 1/4 or 1/5 of my builds. I think it's something related to Netlify, but not sure how to get to the bottom of the problem. I've already updated to v4 a while back and I can't really revert back to v3

**Comment by stevepepple** [comment]
I'm also seeing this issue on Netlify with any version of Gatsby, most recently `@4.7.1`
I emailed Netlify support about this issue, and they said it's because of memory use while processing sharp images. However, I see that the image processing job appears to complete successfully:

```
4:50:53 PM: success run queries in workers - 134.799s - 2149/2149 15.94/s
**4:51:50 PM: success Running gatsby-plugin-sharp.IMAGE_PROCESSING jobs - 787.835s - 1076/1076 1.37/s**
4:51:50 PM: error UNHANDLED EXCEPTION write EPIPE
4:51:50 PM: 
4:51:50 PM: 
4:51:50 PM:   Error: write EPIPE
4:51:50 PM:   
4:51:50 PM:   - child_process.js:841 ChildProcess.target._send
4:51:50 PM:     internal/child_process.js:841:20
4:51:50 PM:   
4:51:50 PM:   - child_process.js:712 ChildProcess.target.send
4:51:50 PM:     internal/child_process.js:712:19
4:51:50 PM:   
4:51:50 PM:   - index.js:291 WorkerPool.sendMessage
4:51:50 PM:     [site]/[gatsby-worker]/dist/index.js:291:19
4:51:50 PM:   
4:51:50 PM:   - worker-messaging.ts:22 
4:51:50 PM:     [site]/[gatsby]/src/utils/jobs/worker-messaging.ts:22:22

```

**Comment by Skil3e** [comment]
It seem like this problem is back. I tried running a build with gatsby 4.8.0 and while the site was building properly with 4.7.1 now it shows the above error. Even after downgrading to 4.7.1 again the error won't go away.

**Comment by kb1995** [comment]
So I'm running two Gatsby websites, both with latest package upgrades and both running on Netlify.

One of the websites uses the new gatsby-plugin-image while the other one uses a mixture of the new and the old (gatsby-image) plugins. Several of the pages have a large amount of images utilising the gatsby-image which seem to overload the memory capacity of Netlify.

The website where I'm only using gatsby-plugin-image doesn't have this problem at all, while the other website experiences this error about 25% of the builds.

**Comment by stevepepple** [comment]
> I'm also seeing this issue on Netlify with any version of Gatsby, most recently `@4.7.1` I emailed Netlify support about this issue, and they said it's because of memory use while processing sharp images. However, I see that the image processing job appears to complete successfully:
> 
> ```
> 4:50:53 PM: success run queries in workers - 134.799s - 2149/2149 15.94/s
> **4:51:50 PM: success Running gatsby-plugin-sharp.IMAGE_PROCESSING jobs - 787.835s - 1076/1076 1.37/s**
> 4:51:50 PM: error UNHANDLED EXCEPTION write EPIPE
> 4:51:50 PM: 
> 4:51:50 PM: 
> 4:51:50 PM:   Error: write EPIPE
> 4:51:50 PM:   
> 4:51:50 PM:   - child_process.js:841 ChildProcess.target._send
> 4:51:50 PM:     internal/child_process.js:841:20
> 4:51:50 PM:   
> 4:51:50 PM:   - child_process.js:712 ChildProcess.target.send
> 4:51:50 PM:     internal/child_process.js:712:19
> 4:51:50 PM:   
> 4:51:50 PM:   - index.js:291 WorkerPool.sendMessage
> 4:51:50 PM:     [site]/[gatsby-worker]/dist/index.js:291:19
> 4:51:50 PM:   
> 4:51:50 PM:   - worker-messaging.ts:22 
> 4:51:50 PM:     [site]/[gatsby]/src/utils/jobs/worker-messaging.ts:22:22
> ```

I did want to share that I reduce the amount and size of images that we were processing (from Wordpress). In particular, I was able to consolidate duplicate images from SEO and other fields. I also make sure we didn't have any large images in terms of resolution width and bytes. After that our build time was drastically reduced by half and the Gatsby 4 builds in Netlify no longer time out.

**Comment by MuhammadHadiofficial** [comment]
Still facing the same issue.
![image](https://user-images.githubusercontent.com/31585104/155687702-12a90d74-491c-4d0f-acab-e6dcdd91d097.png)
Any solution?

**Comment by kb1995** [comment]
@MuhammadHadiofficial Migrate away from gatsby-image

**Comment by MuhammadHadiofficial** [comment]
@kb1995 Hi, I didn't get it what are other options?

**Comment by stevepepple** [comment]
> @MuhammadHadiofficial Migrate away from gatsby-image

That's not really a solution, since Gatsby image is a core part of what makes Gatsby nice.

**Comment by Skil3e** [comment]
I ended up resizing and resaving my images and it builds successfully now. What seems weird to me is that I have another site with a lot more images that never had this issue. For reference the site that failed had a total of 25 images.

**Comment by MuhammadHadiofficial** [comment]
@Skil3e can you tell me the actual difference between the size of your image before and after resizing was the size difference much larger?

**Comment by stevepepple** [comment]
> @MuhammadHadiofficial Migrate away from gatsby-image

That's not really a solution, since Gatsby image is a core part of what makes Gatsby nice.

**Comment by MuhammadHadiofficial** [comment]
I'm using gatsby-plugin-image already on previous plugin we didn't had this issue after migrating to gatsbyv4 and new image plugin we started to face these memory related issues.

**Comment by stevepepple** [comment]
> > > @MuhammadHadiofficial Migrate away from gatsby-image
> > 
> > 
> > That's not really a solution, since Gatsby image is a core part of what makes Gatsby nice.
> 
> You can migrate from gatsby-image (a deprecated API) to gatsby-plugin-image (new API). You can still continue using Gatsby for what makes it nice

Oh, I see what you mean. That said, I believe the same memory issues happen with new API (gatsby-plugin-image). I'd not seen the issue after reducing image size, but a few new Wordpress images triggered this issue just the other day. We removed all instances of `gatsby-image`

**Comment by MuhammadHadiofficial** [comment]
@stevepepple if you removed the gatsbyimagedats?
Then how you are rendering the images?

**Comment by stevepepple** [comment]
> @stevepepple if you removed the gatsbyimagedats? Then how you are rendering the images?

It's a little confusing. I removed `gatsby-image` but I'm using `gatsby-plugin-image`

**Comment by panzacoder** [comment]
Hi @stevepepple and @MuhammadHadiofficial you should reference the migration doc: https://www.gatsbyjs.com/docs/reference/release-notes/image-migration-guide/

GatsbyImage originally came pre-baked into Gatsby v1-2, but was later deprecated in favor of the plugin approach, which has a simpler API and some additional functionality (including somewhat better performance for page visitors). 

Unfortunately I do not believe that it is a solution to this particular problem.

**Comment by MuhammadHadiofficial** [comment]
> Hi @stevepepple and @MuhammadHadiofficial you should reference the migration doc: https://www.gatsbyjs.com/docs/reference/release-notes/image-migration-guide/
> 
> GatsbyImage originally came pre-baked into Gatsby v1-2, but was later deprecated in favor of the plugin approach, which has a simpler API and some additional functionality (including somewhat better performance for page visitors). 
> 
> Unfortunately I do not believe that it is a solution to this particular problem.

Hi @panzacoder , I've already migrated to latest Gatsby version4 and latest Gatsby image plugin and ran the migration script also.
Problem is this new plugin uses more resources as compared to what netlify provides and it goes out of memory and process gets killed.

**Comment by panzacoder** [comment]
> Hi @panzacoder , I've already migrated to latest Gatsby version4 and latest Gatsby image plugin and ran the migration script also. Problem is this new plugin uses more resources as compared to what netlify provides and it goes out of memory and process gets killed.

Thanks for clarifying the state of your project. I have also had the same issues with Netlify, and in general Gatsby v4 (I actually ran into memory issues running on Gatsby Cloud as well.

I'm assuming there are deficiencies in our build step that need to be addressed, but I don't have time to figure that our right now, so I migrated down to Gatsby v3. I believe there was only one plugin I had to update to get stable builds again, but YMMV.

I am hoping to have time to dig into our performance build issues sooner-than-later but right now our marketing team requires focus elsewhere. 

What's frustrating for me overall is that this seems to be a pretty big issue (memory usage in build step) but there's not a clear path forward on figuring out what is causing it, whether it's a framework issue, a plugin issue, or a custom code issue.

**TL;DR: Revert to Gatsby v3 until you have time/budget to diagnose your build performance**

**Comment by stevepepple** [comment]
Thanks @panzacoder. Why is this issue closed though? I don't believe downgrading is an adequate work around for many who need to use the latest version of Gatsby.

**Comment by byebyers** [comment]
This issue should not be closed. Image processing works on my local environment but not Netlfiy. Something that started happening after updating Gatsby.

**Comment by hari-s** [comment]
I was facing this issue while deploying with netlify with gatsby version 4.11.0,
it worked for me while I down graded gatsby version to **4.1.3**

**Comment by jdahdah** [comment]
This issue recently started happening to me across three client projects. One of which was already upgraded to Gatsby v4 before it started and can't be rolled back to v3. The other one is waiting for a fix so that the upgrade can be completed. But the most clear demonstration of this issue is the most recent project, which I started about a week ago (so of course it uses v4). It only has about 14 images that are a few hundred KBs each (but reused in a few places at different sizes), across 20 pages . Build are failing left and right and only sporadically work. It's extremely unreliable.

There is a major issue going on with Gatsby v4 and Netlify. When will it be taken seriously?

```
success write out requires - 0.003s
6:50:51 PM: success Building production JavaScript and CSS bundles - 25.772s
6:51:12 PM: success Building HTML renderer - 20.601s
6:51:12 PM: success Execute page configs - 0.069s
6:51:12 PM: success Caching Webpack compilations - 0.001s
6:51:15 PM: success run queries in workers - 2.929s - 29/29 9.90/s
6:51:28 PM: success Running gatsby-plugin-sharp.IMAGE_PROCESSING jobs - 15.818s - 40/40 2.53/s
6:51:28 PM: error UNHANDLED EXCEPTION write EPIPE
6:51:28 PM: 
6:51:28 PM: 
6:51:28 PM:   Error: write EPIPE
6:51:28 PM:   
6:51:28 PM:   - child_process:866 ChildProcess.target._send
6:51:28 PM:     node:internal/child_process:866:20
6:51:28 PM:   
6:51:28 PM:   - child_process:739 ChildProcess.target.send
6:51:28 PM:     node:internal/child_process:739:19
6:51:28 PM:   
6:51:28 PM:   - index.js:298 WorkerPool.sendMessage
6:51:28 PM:     [repo]/[gatsby-worker]/dist/index.js:298:19
6:51:28 PM:   
6:51:28 PM:   - worker-messaging.ts:22 
6:51:28 PM:     [repo]/[gatsby]/src/utils/jobs/worker-messaging.ts:22:22
6:51:28 PM:   
6:51:28 PM: 
6:51:28 PM: not finished Merge worker state - 0.044s
6:51:29 PM: ​
6:51:29 PM: ────────────────────────────────────────────────────────────────
6:51:29 PM:   "build.command" failed                                        
6:51:29 PM: ────────────────────────────────────────────────────────────────
6:51:29 PM: ​
6:51:29 PM:   Error message
6:51:29 PM:   Command failed with exit code 1: gatsby build (https://ntl.fyi/exit-code-1)
6:51:29 PM: ​
6:51:29 PM:   Error location
6:51:29 PM:   In build.command from netlify.toml:
6:51:29 PM:   gatsby build
```

**Comment by wktk** [comment]
In the case of the issue with v4.8+ and Netlify, #35055 is currently ongoing.

**Comment by MuhammadHadiofficial** [comment]
@panzacoder  any updates on the issue?

**Comment by andrelandgraf** [comment]
I am on Gatsby v^4.11.2. The issue **only** came up for me after I changed an image in my repository. I am seeing this issue now on my Netlify build as well and it seems to be connected to gatsby-plugin-image. New image had a file size of 24 MB. Reducing the file size to 2MB fixed the issue for me! 👍 

I should probably not use huge images anyways 😃  but since Gatsby's image logic is handling optimization for me, I didn't pay much attention tbh. Hope this helps others!

**Comment by StanLepunK** [comment]
Same isssue when I deploy on Netlify with
```
"gatsby": "^4.12.1",
"gatsby-plugin-image": "^2.12.1",
"gatsby-transformer-sharp": "^4.12.1",
"gatsby-plugin-sharp": "^4.12.1",
```

**Comment by ascorbic** [comment]
Hey. This is a duplicate of #35055 , which now has a fix. If you update to `gatsby@4.13.1` then it should fix the problem.
